client/src/components/horizontal-pane/horizontal-pane.tsx

```diff
@@ -1,27 +1,36 @@
-import clsx from 'clsx'
-import React, { useState, useEffect } from 'react'
+import React, { useState, useEffect, useCallback } from 'react'
 import { DraggableCore } from 'react-draggable'
+import usePersistedSetState from 'hooks/use-persisted-state'
 
-export type HorizontalPaneProps = {
-  className?: string
-  children?: React.ReactNode
+type HorizontalPaneProps = {
+  render: (props: {
+    paneWidth: number
+    isFullScreen?: boolean
+    dragHandle: React.ReactNode
+    toggleFullScreen?: () => void
+  }) => React.ReactElement
+  paneName: string
   initialWidth?: number
   minConstraint?: number
   maxConstraint?: number
   buffer?: number
 }
 
 export default function HorizontalPane({
-  className,
-  children,
+  render,
+  paneName,
   initialWidth = 240,
   minConstraint = 0,
   maxConstraint,
   buffer,
 }: HorizontalPaneProps) {
-  const [sidebarWidth, setSidebarWidth] = useState(initialWidth)
+  const [paneWidth, setPaneWidth] = usePersistedSetState<number>(
+    paneName,
+    initialWidth,
+  )
   const [dragDirection, setDragDirection] = useState<string | null>(null)
 
+  const isFullScreen = paneWidth === 0
```

Comment:
Even though the logic is correct, but this line seems really confusing. Generally `fullScreen` means that the `paneWidth` would full. So we should rename the `paneWidth` to specify that it is meant for the first item.